nautobot/docs/user-guides/git-data-source.md

```diff
@@ -0,0 +1,269 @@
+
+
+## Using Git Data Sources
+This section will focus on examples and use the `user-guide` branch on the demo-git-datasources repo: `https://github.com/nautobot/demo-git-datasource`.
+
+### Export Templates
+
+[Export Templates](../models/extras/exporttemplate.md) allow a user to export Nautobot objects based on a custom template.  Export templates can change over time depending on the needs of a user.  Allowing export templates to reference a Git repo makes managing templates easier.
+
+A template can be used to put objects into a specific format for ingestion into another system, tool, or report.  It is possible that different templates are needed depending on specific users or teams.  This can lead to sprawl of export templates.  To keep accurate templates synced with Nautobot the Git Data Sources extensibility feature can be used.
+
+#### Step 1
+Navigate to the Data Sources Git integration. **Extensibility -> Git Repositories**.
+
+![](./images/git-data-source_1.png)
+
+Click [+] or [Add]
+
+That loads a default page to add a repository.
+
+![](./images/git-data-source_2.png)
+
+!!! note
+    By default only config contexts, export templates, and jobs are available resource types.  Others may be added when specific plugins are used.
+
+#### Step 2
+Fill out the details for the Git repository. More information on the inputs can be found in the [fields section](#repository-details).
+
+![](./images/git-data-source_3.png)
+
+As soon as you click on **Create**, Nautobot will clone and sync the repository and provide status of the job.
+
+![](./images/git-data-source_4.png)
+![](./images/git-data-source_6.png)
+
+The repository will now be displayed on the main Git Repository page.
+
+![](./images/git-data-source_5.png)
+
+Once the repository is synced each template will now be available in the Export Templates section.  **Extensibility -> Export Templates**.
+
+![](./images/git-data-source_7.png)
+
+> !!! note
+    If the templates don't populate, make sure the Git directory is named `export_templates` and the sub-directory and sub-sub-directory names correctly match the Nautobot `content type`.
+
+Example below:
+```
+▶ tree export_templates 
+export_templates
+└── dcim
+    └── device
+        ├── markdown_export.md
+        ├── text_export.txt
+        └── yaml_export.yml
+
+2 directories, 3 files
+```
+
+#### Step 3
+Now that the export templates have been loaded into Nautobot they can be utilized as normal.  For example navigate to **Devices -> Devices** and click on **Export** in the top right corner, the dropdown will now include the templates loaded from the Git repository.
+
+The power of having export templates utilizing the Git integration comes with the native source control features that Git comes with.  To illustrate a simple Git sync within Nautobot assume the following template needs to be updated.
+
+Filename: `/export_templates/dcim/device/yaml_export.yml`
+
+Current contents:
+```jinja
```

Comment:
FWIW, to enable `jinja` syntax highlighting, I think you'll need to add it in `mkdocs.yml`:

```yaml
    hljs_languages:
        - jinja
        - yaml
```

Reply:
@glennmatthews  Should we do this? Or just use standard code block?  I don't think the Jinja2 highlighting is a requirement here.

Reply:
I'm in favor of it - it makes the readability a bit better.